runtime/compiler/runtime/J9Profiler.hpp

```diff
@@ -612,6 +663,9 @@ class TR_MethodBranchProfileInfo
    uint32_t _initialBlockFrequency;
    };
 
+// To be used for checking if _counterDerivationInfo[i] is a bit vector or not
+#define IS_VALID_BIT_VECTOR(cdi) (!((uintptr_t)cdi & 0x1))
```

Comment:
I do not think so `IS_VALID_BIT_VECTOR` is the correct name here. 
IIRC low tagged entry means we need to just get the counter from single block to either add or subtract, otherwise, it should hold the array of bit vector representing counters from blocks added and subtracted together to get the frequency. I am not sure if this is the correct check. Take a look at the comment here,
https://github.com/eclipse/openj9/blob/b2400f9240b80a0a41d314ddc5768a42915a69ca/runtime/compiler/runtime/J9Profiler.hpp#L660-L663

Reply:
@r30shah  If I understand this correctly the comment says if the entry has is low tagged it would have some integer value, otherwise it would be pointing to `TR_BitVector`.
`IS_VALID_BIT_VECTOR` returns true if the entry is not low tagged, which seems to be in-sync with the comments.
It is missing a NULL check, I can add that to the condition. Is there anything else you are pointing to?

Reply:
Yeah, I think missing NULL check should be there. 
 What I meant originally is the name `IS_VALID_BIT_VECTOR` , the name seems more generic but the check is internal to BlockFrequencyInfo, so IMO it should either a name that justifies the internal use, or a private static method (To make code more readable) instead of the macro